src/components/modal.js

```diff
@@ -0,0 +1,166 @@
+(() => ({
+  name: 'Modal',
+  type: 'CONTAINER_COMPONENT',
+  allowedTypes: ['BODY_COMPONENT', 'CONTENT_COMPONENT'],
+  orientation: 'HORIZONTAL',
+  jsx: (
+    <div className={B.env === 'dev' ? classes.panel : ''}>
+      {(() => {
+        const { env } = B;
+        const { Modal } = window.MaterialUI.Core;
+
+        const isEmpty = children.length === 0;
+        const isDev = env === 'dev';
+        const isPristine = isEmpty && isDev;
+
+        const [open, setOpen] = React.useState(false);
+
+        useEffect(() => {
+          B.defineFunction('handleOpenModal', e => {
+            e.preventDefault();
+
+            setOpen(true);
+          });
+
+          B.defineFunction('handleCloseModal', e => {
+            e.preventDefault();
+
+            setOpen(false);
+          });
+        }, []);
+
+        return (
+          <div>
+            {isDev && options.isVisibleInDev ? (
+              <div
+                className={[
+                  classes.content,
+                  isEmpty ? classes.empty : '',
+                  isPristine ? classes.pristine : '',
+                ].join(' ')}
+              >
+                {isPristine ? 'Panel' : children}
+              </div>
+            ) : (
+              ''
+            )}
+
+            <Modal
+              open={open}
+              onClose={() => setOpen(false)}
+              aria-labelledby="simple-modal-title"
+              aria-describedby="simple-modal-description"
+            >
+              <div className={classes.modal}>{children}</div>
```

Comment:
This component is always shown on top of the window. I'm not sure if that is what we want.